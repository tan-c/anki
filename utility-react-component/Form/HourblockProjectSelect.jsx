import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';
import { Map } from 'immutable';
import { projectsByCategoryNameSelector } from 'utility-redux/project';

export class ProjectSelect extends React.Component {
  // constructor(props, context) {
  //   super(props, context);
  // }
  //
  // componentWillReceiveProps(nextProps) {
  //   this.setState({
  //   });
  // }

  render() {
    const {
      projectsByCategoryName,
      onChangeEvent, value, color, className,
    } = this.props;

    return (
      <select
        type="text"
        name="project"
        className={className}
        onChange={onChangeEvent}
        value={value}
        style={{ borderLeft: `5px solid ${color}` }}
      >
        <option value="" />
        {projectsByCategoryName.keySeq().map(categoryName => (
          <optgroup key={categoryName} label={categoryName}>
            {projectsByCategoryName.get(categoryName).valueSeq().map(project => (
              <option key={project.get('_id')} value={`${project.get('_id')}`}>
                {project.get('name')}
              </option>))}
          </optgroup>
        ))}

      </select>
    );
  }
}

ProjectSelect.defaultProps = {
  projectsByCategoryName: Map(),
  // sortedProjectList: Map(),
  className: '',
  value: '',
  color: 'transparent',
  onChangeEvent: () => { },
};

ProjectSelect.propTypes = {
  projectsByCategoryName: PropTypes.object,
  // sortedProjectList: PropTypes.object,
  value: PropTypes.string,
  className: PropTypes.string,
  color: PropTypes.string,
  onChangeEvent: PropTypes.func,
};

function mapStateToProps(state, ownProps) {
  return {
    projectsByCategoryName: projectsByCategoryNameSelector(state),
    // sortedProjectList: projectsByIdSortedSelector(state),
    className: ownProps.className,
    value: ownProps.value,
    color: ownProps.color,
    onChangeEvent: ownProps.onChangeEvent,
  };
}

export default connect(mapStateToProps)(ProjectSelect);
