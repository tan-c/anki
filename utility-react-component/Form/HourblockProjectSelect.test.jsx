import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { ProjectSelect } from './HourblockProjectSelect';

function setup() {
  const props = {
  };

  return shallow(<ProjectSelect {...props} />);
}

describe('ProjectSelect Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find('select')).toHaveLength(1);
    });
  });
});
