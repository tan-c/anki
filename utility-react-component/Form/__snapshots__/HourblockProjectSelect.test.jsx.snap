// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSelect Unit Tests when normally set up renders correctly 1`] = `
<select
  className=""
  name="project"
  onChange={[Function]}
  style={
    Object {
      "borderLeft": "5px solid transparent",
    }
  }
  type="text"
  value=""
>
  <option
    value=""
  />
</select>
`;
